test/common/stats/thread_local_store_test.cc

```diff
@@ -189,6 +246,251 @@ TEST_F(StatsThreadLocalStoreTest, BasicScope) {
   // Includes overflow stat.
   EXPECT_CALL(*this, free(_)).Times(5);
 }
+TEST_F(StatsThreadLocalStoreTest, BasicSingleHistogramMerge) {
+  InSequence s;
+  store_->initializeThreading(main_thread_dispatcher_, tls_);
+
+  Histogram& h1 = store_->histogram("h1");
+  EXPECT_EQ("h1", h1.name());
+
+  std::vector<uint64_t> h1_cumulative_values;
+  std::vector<uint64_t> h2_cumulative_values;
+  std::vector<uint64_t> h1_interval_values;
+  std::vector<uint64_t> h2_interval_values;
+
+  EXPECT_CALL(sink_, onHistogramComplete(Ref(h1), 0));
+  h1.recordValue(0);
+  h1_cumulative_values.push_back(0);
+  h1_interval_values.push_back(0);
```

Comment:
I would turn this repeated 4-line stanza a helper method, e.g.

  expectCallAndAccmumute(expected_call, record_value, cumulative_value, interval_value) {
     ...
  }

...
  {
    expectCallAndAccmumute(43, 43, 43, 43);
    expectCallAndAccmumute(41, 41, 41, 41);
...

It's interesting that when you factor out all this repeated stuff you might see patterns more easily.  I imagine that in other sub-tests the 4 args will not all be the same?

Reply:
ok. Good pattern. I have even simplified further by 
`expectCallAndAccumulate(Histogram& histogram, uint64_t record_value)` as `record_value` is what we push to interval as well as cumulative histograms